openshift-201/resource-mgmt.md

```diff
@@ -104,45 +104,48 @@ spec:
     - Ingress
 ```
 
-Next, create a new deployment for load testing. This will deploy an httpd container and use the `ab` (apache benchmark) command to generate traffic to a URL and then print a summary. Then the pod will stop. If you update the environment variables for the deployment, that will trigger a pod redeployment to run the load test again. Update the deployment below with the URL to your nginx web server under the `SERVICE_HOST` variable.
+Next, create a new 'job' for load testing. This will deploy an httpd container and use the `ab` (apache benchmark) command to generate traffic to a URL and then print a summary. Then the pod will stop. If you update the environment variables for the deployment, that will trigger a pod redeployment to run the load test again. Update the deployment below with the URL to your nginx web server under the `SERVICE_HOST` variable. 
+
+**Note: this differs from the video where a Deployment is shown, we'll use a Job**
 
 ```yaml
-kind: Deployment
-apiVersion: apps/v1
+apiVersion: batch/v1
+kind: Job
 metadata:
-  name: load-test
-  labels:
-    app: load-test
+  name: load-test-job
 spec:
-  replicas: 1
-  selector:
-    matchLabels:
-      app: load-test
   template:
     metadata:
-      annotations:
       labels:
         app: load-test
     spec:
       containers:
         - name: load-test
           image: registry.access.redhat.com/rhscl/httpd-24-rhel7
           env:
-          - name: SERVICE_HOST
-            value: "hello-world-lab1.apps.ocp4.example.com"
-          - name: SERVICE_PORT
-            value: "443"
-          - name: REQUESTS
-            value: "500000"
-          - name: CONCURRENCY
-            value: "20"
+            - name: SERVICE_HOST
+              value: "hello-world-lab1.apps.ocp4.example.com"
+            - name: SERVICE_PORT
+              value: "443"
+            - name: REQUESTS
+              value: "500000"
+            - name: CONCURRENCY
+              value: "20"
           command: ["/opt/rh/httpd24/root/usr/bin/ab"]
-          args: ["-dSrk", "-c $(CONCURRENCY)", "-n $(REQUESTS)", "https://$(SERVICE_HOST):$(SERVICE_PORT)/index.html"]
+          args: ["-dSrk", "-c", "$(CONCURRENCY)", "-n", "$(REQUESTS)", "https://$(SERVICE_HOST):$(SERVICE_PORT)/index.html"]
+          resources:
+            requests:
+              memory: "256Mi"  
+              cpu: "100m"       
+            limits:
+              memory: "512Mi"  
+              cpu: "200m"       
+      restartPolicy: Never
+  backoffLimit: 1
+  ttlSecondsAfterFinished: 3600
```

Comment:
```suggestion
  activeDeadlineSeconds: 3660
```
Job should have a `activeDeadlineSeconds` to control how long it runs. `ttlSecondsAfterFinished` just causes the job to be deleted after it has completed to clean it up. Also putting it slightly longer than the Timeout passed to `ab`.